sdks/java/core/src/main/java/org/apache/beam/sdk/util/state/StateSpecs.java

```diff
@@ -44,6 +45,11 @@
   private StateSpecs() {}
 
   /** Create a simple state spec for values of type {@code T}. */
+  public static <T> StateSpec<Object, ValueState<T>> value() {
+    return value(null);
```

Comment:
I would say that the contract on the `value(Coder)` factory method is that the coder is non-null (and it could have a `checkArgument` that enforces it). Here, you should just directly call the constructor with `null` instead.